### ISSUE TITLE
Pillow 8.1.2 incompatible with Python 3.10

### ISSUE THREAD

**Issue by alkenes** [servo/servo#28623]
OS: Fedora 35

I have two reports to make in regards to Servo on Fedora 35, first bootstrap is reporting that fedora is unsupported and I should open a ticket, let me know if I should open a separate ticket.

Second issue is a dependency issue with python 3.10 (default installation in fedora 35), Pillow 8.1.2 is not compatible. Based on the support matrix they provide [here](https://pillow.readthedocs.io/en/latest/installation.html) I incremented the version to 8.3.2 but it could go as high as 8.4 without loosing support for any python versions. 

~~Is python 3.10 support something that you would like to bring in? If so I can work on a PR to update the code base to support it, currently I'm seeing additional issues around the collections library, possibly more issues to surface.~~ I didn't notice I was digging around in mach's code at first, looks like the only change in servo code that is needed for 3.10 is this dependency being bumped.

For anyone trying to finding this via google the dependency issue was producing 'The headers or library files could not be found for jpeg, a required dependency when compiling Pillow from source.' as well as 'ERROR: Failed building wheel for pillow' and the fix can be applied in <servo repo>/tests/wpt/web-platform-tests/tools/wptrunner/requirements.txt. But there will be more fixes needed to work with 3.10.

**Comment by jdm** [comment]
What does https://github.com/servo/servo/blob/9693111c9744407730d7fc971a33800a831d575e/python/servo/bootstrap.py#L332 return for you? Fedora is supposed to be supported according to that code.

And yes, bumping dependencies to support newer python versions sounds fine.

**Comment by jdm** [comment]
Oh, if it comes from https://github.com/servo/servo/blob/0d0cfd030347ab0711b3c0607a9ee07ffe7124cf/tests/wpt/web-platform-tests/tools/wptrunner/requirements.txt#L7 then that's more of an issue. That code is from upstream, but we haven't been syncing with upstream for a lot of months.

**Comment by alkenes** [comment]
I pulled that branch down, I'll take a look and see if I can help with that at all. First step is going to be updating mozjs for 3.10 apparently.

I see the problem with bootstrap, distrib is coming back as 'Fedora Linux' not 'Fedora' like the script is expecting. Adding that as an option in the block here

https://github.com/servo/servo/blob/9693111c9744407730d7fc971a33800a831d575e/python/servo/bootstrap.py#L380

fixes the issue. I did notice there's a comment `# Fixme: we should allow checked/supported versions only` above it, still valid or should I get rid of that while I'm in there?
